prog/rand.go

```diff
@@ -754,3 +761,84 @@ func (a *CsumType) generate(r *randGen, s *state) (arg Arg, calls []*Call) {
 	// Filled at runtime by executor.
 	return MakeConstArg(a, 0), nil
 }
+
+// Finds a compatible resource with the type `t` and the calls that initialize that resource.
+func resourceCentric(t *ResourceType, s *state, r *randGen) (resource *ResultArg, calls []*Call) {
+	if len(s.corpus) == 0 {
+		return nil, nil
+	}
+
+	corpusResources := make(map[*ResultArg](map[*Call]bool))
+	for _, p := range s.corpus {
+		addCompatibleResources(p.Clone(), t.TypeName, r, corpusResources)
```

Comment:
I think you should:
1. Find all compatible resources within the corpus.
2. Choose one of them randomly.
3. And only now extract calls that refer to this resource.

Reply:
BTW, it might be possible to optimize steps 1 and 2 by choosing some pseudo random order of iterating throughout the programs in the corpus, and then just grabbing a random resource from the first program that contains compatible resources.